elftools/elf/relocation.py

```diff
@@ -89,6 +90,54 @@ def iter_relocations(self):
             yield self.get_relocation(i)
 
 
+class RelocationSection(Section, RelocationTable):
+    """ ELF relocation section. Serves as a collection of Relocation entries.
+    """
+    def __init__(self, header, name, elffile):
+        Section.__init__(self, header, name, elffile)
+        RelocationTable.__init__(self, self.stream, self.elffile,
+            self['sh_offset'], self['sh_size'], header['sh_type'] == 'SHT_RELA')
+
+        elf_assert(header['sh_type'] in ('SHT_REL', 'SHT_RELA'),
+            'Unknown relocation type section')
+        elf_assert(header['sh_entsize'] == self.entry_size,
+            'Expected sh_entsize of %s section to be %s' % (
+                header['sh_type'], self.entry_size))
+
+
+def get_dynamic_reloc_tables(stream, elffile, dynamic):
```

Comment:
Is there a case where we need a different `stream` than the one from the `elffile` parameter? If not, it would probably be cleaner to just pass `elffile` and access the stream within this function.

I'm inclined to say this would also be the case for the `dynamic` parameter, so we could do all checking inside this function and just pass in an `ELFFile`.

Reply:
I think previously `stream` was a separate parameter to `RelocationSection`'s constructor. At some point this changed, so I corrected `RelocationSection`'s signature but left `stream` as a separate parameter. I can correct everything to just remove the `stream` parameter and rely on `elffile`.

`dynamic`, on the other hand, might be either a section or a segment, and I don't want to preclude the API user from specifying it. If it helps, maybe I could make it default to `None` which triggers some automatic behaviour (e.g. default to the DYNAMIC segment).

Reply:
It's still not fully consistent between segments and sections (the individual  `Segment` constructors get passed a `stream`, and some an `ELFFile` as well - `DynamicSegment` is one of those) but I'd prefer dropping the `stream` parameter here.

For `dynamic`, you are right, with either a segment or a section passed in this is more flexible. Adding a default parameter would probably only worsen readability so maybe it's best to just leave it as a mandatory parameter.